neurosym/examples/near/search_graph.py

```diff
@@ -7,31 +9,79 @@
     FilterEdgesGraph,
     LimitEdgesGraph,
 )
-from neurosym.types.type import Type
+from neurosym.types.type import ArrowType, AtomicType, ListType, TensorType, Type
+from neurosym.utils.logging import log
 
 
 class FilterUnexpandableNodes(FilterEdgesGraph):
     """
     Represents a search graph that removes nodes that cannot be expanded.
 
-    Currently does nothing. This is a placeholder for future functionality.
+    Removes edges to partial programs that will be too deep.
     """
 
+    def __init__(self, graph: SearchGraph, max_depth: int):
+        super().__init__(graph=graph)
+        self.max_depth = max_depth
+
+    def depth2go(self, node: SExpression) -> int:
+        """
+        Calculates a lower bound estimate of the depth2go. For a partial program,
+        this is approximated by the minimum number of replacement nodes needed for
+        each type of hole.
+
+        :param node: The node to estimate depth2go for.
+        :returns height: The minimum possible height of the node if partial programs
+            were filled.
+        """
+        height = 1
+        if isinstance(node, Hole):
+            # @TODO: Better heuristic than this.
+            match node.twe.typ:
```

Comment:
Not really a fan of this. I had to do this in `neurosym.examples.near.neural_dsl:get_shape` as well and it breaks functionality if someone wants to add more types in the future.

Reply:
there's Type.depth, which you might use instead?

Reply:
Looked into this. The core issue in this piece of code isn't depth calculation, it's iterating over all “registered” types. Anyway, I think this is another issue entirely. Merging these changes for now.